### ISSUE TITLE
Appveyor is flaky

### ISSUE THREAD

**Issue by lithp** [citusdata/citus#2136]
`multi_real_time_transaction` sometimes fails, claiming every line has changed, and sometimes succeeds. It seems to happen randomly (restarting the build can cause it to pass).

Here's an example pass: https://ci.appveyor.com/project/lithp/citus/build/1.0.800-pg11_compat?fullLog=true
And an example fail: https://ci.appveyor.com/project/lithp/citus/build/1.0.798-pg11_compat?fullLog=true

They seem to be the exact same until the end.

**Comment by lithp** [comment]
Maybe we pass or fail depending on which build worker handles the build?

**Comment by lithp** [comment]
It's apparently possible to remote into the worker to diagnose problems: https://www.appveyor.com/docs/how-to/rdp-to-build-worker/

**Comment by lithp** [comment]
I added a primitive hexdump to the test output. The line endings are different on `multi_real_time_transaction`, but they're also different on `with_basics`!

```
$s = Get-Content -Path .\expected\multi_real_time_transaction.out -Raw
$s.substring(0, 100) | Format-Hex
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   53 45 54 20 63 69 74 75 73 2E 6E 65 78 74 5F 73  SET citus.next_s
00000010   68 61 72 64 5F 69 64 20 54 4F 20 31 36 31 30 30  hard_id TO 16100
00000020   30 30 3B 0A 43 52 45 41 54 45 20 53 43 48 45 4D  00;.CREATE SCHEM
00000030   41 20 6D 75 6C 74 69 5F 72 65 61 6C 5F 74 69 6D  A multi_real_tim
00000040   65 5F 74 72 61 6E 73 61 63 74 69 6F 6E 3B 0A 53  e_transaction;.S
00000050   45 54 20 73 65 61 72 63 68 5F 70 61 74 68 20 3D  ET search_path =
00000060   20 27 6D 75                                       'mu            
$s = Get-Content -Path .\results\multi_real_time_transaction.out -Raw
$s.substring(0, 100) | Format-Hex
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   53 45 54 20 63 69 74 75 73 2E 6E 65 78 74 5F 73  SET citus.next_s
00000010   68 61 72 64 5F 69 64 20 54 4F 20 31 36 31 30 30  hard_id TO 16100
00000020   30 30 3B 0D 0A 43 52 45 41 54 45 20 53 43 48 45  00;..CREATE SCHE
00000030   4D 41 20 6D 75 6C 74 69 5F 72 65 61 6C 5F 74 69  MA multi_real_ti
00000040   6D 65 5F 74 72 61 6E 73 61 63 74 69 6F 6E 3B 0D  me_transaction;.
00000050   0A 53 45 54 20 73 65 61 72 63 68 5F 70 61 74 68  .SET search_path
00000060   20 3D 20 27                                       = '            
$s = Get-Content -Path .\expected\with_basics.out -Raw
$s.substring(0, 100) | Format-Hex
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   2D 2D 20 54 65 73 74 20 74 68 65 20 62 61 73 69  -- Test the basi
00000010   63 20 43 54 45 20 66 75 6E 63 74 69 6F 6E 61 6C  c CTE functional
00000020   69 74 79 20 61 6E 64 20 65 78 70 65 63 74 65 64  ity and expected
00000030   20 65 72 72 6F 72 20 6D 65 73 73 61 67 65 73 0A   error messages.
00000040   43 52 45 41 54 45 20 54 59 50 45 20 78 79 20 41  CREATE TYPE xy A
00000050   53 20 28 78 20 69 6E 74 2C 20 79 20 69 6E 74 29  S (x int, y int)
00000060   3B 0A 53 45                                      ;.SE            
$s = Get-Content -Path .\results\with_basics.out -Raw
$s.substring(0, 100) | Format-Hex
           00 01 02 03 04 05 06 07 08 09 0A 0B 0C 0D 0E 0F
00000000   2D 2D 20 54 65 73 74 20 74 68 65 20 62 61 73 69  -- Test the basi
00000010   63 20 43 54 45 20 66 75 6E 63 74 69 6F 6E 61 6C  c CTE functional
00000020   69 74 79 20 61 6E 64 20 65 78 70 65 63 74 65 64  ity and expected
00000030   20 65 72 72 6F 72 20 6D 65 73 73 61 67 65 73 0D   error messages.
00000040   0A 43 52 45 41 54 45 20 54 59 50 45 20 78 79 20  .CREATE TYPE xy 
00000050   41 53 20 28 78 20 69 6E 74 2C 20 79 20 69 6E 74  AS (x int, y int
00000060   29 3B 0D 0A                                      );..            
```

([from here](https://ci.appveyor.com/project/lithp/citus/build/1.0.862-debug-appveyor))

**Comment by lithp** [comment]
Maybe this is some kind of race condition?

**Comment by onderkalaci** [comment]
#2107 has the same issue. Merging to the master with failures

**Comment by onderkalaci** [comment]
> Maybe this is some kind of race condition?

After reading your investigation, I found this [link](https://help.appveyor.com/discussions/problems/1119-allow-changing-git-autocrlf-setting) which seems something that we're already doing.

There is also this [link](https://github.com/code-cracker/code-cracker/commit/270c5d3fcce9dc74d772bb573672628d71ed916e#diff-fc723d30b02a4cca7a534518111c1a66) where some project that had a similar issue fixes it.

Could we have something similar? For example, do you know that does the following mean that is in Citus' `.gitattributes` file:
```
# Test output files that contain extra whitespace
*.out                                   -whitespace
```

**Comment by lithp** [comment]
I tried running `unix2dos` and `dos2unix` on my Windows box and was unable to get the tests to fail. `pg_regress` [passes the correct flags](https://github.com/postgres/postgres/blob/master/src/test/regress/pg_regress.c#L67) to `diff` to tell it to ignore that kind of thing. This is more strong evidence that the line endings aren't the part which is messing with us.

**Comment by lithp** [comment]
I tried adding [RDP](https://www.appveyor.com/docs/how-to/rdp-to-build-worker/) to the worker so I'd be able to figure out why diff is failing, then I was reminded that the tests are flaky [and the tests passed](https://ci.appveyor.com/project/lithp/citus/build/1.0.1093-debug-appveyor?fullLog=true).

**Comment by lithp** [comment]
I've diagnosed the problem and fixed it with #2159.
